app/forms/claims/claim/mentors_form.rb

```diff
@@ -17,6 +17,14 @@ def persist
     claim.save!
   end
 
+  def edit_back_path
+    if claim.reviewed
```

Comment:
Actually, a nitpick, but could we add a `?` here?

```suggestion
    if claim.reviewed?
```

and the same for others